apps/rig_api/lib/rig_api/router.ex

```diff
@@ -37,4 +37,29 @@ defmodule RigApi.Router do
     pipe_through(:api)
     get("/", HealthController, :check_health)
   end
+
+  scope "/swagger-ui" do
+    forward("/", PhoenixSwagger.Plug.SwaggerUI,
+      otp_app: :rig_api,
+      swagger_file: "rig_api_swagger.json"
+    )
+  end
+
+  def swagger_info do
+    %{rig: rig_version} = versions()
```

Comment:
Please replace this with `Application.spec(:rig, :vsn)`. The "version" file might not be present in a release. The reason for the `Code.eval_file` thing in the Mix files is that it is currently not possible to globally set the project version for an umbrella project, but that's just Mix and it's not meant to be used in production code.

Reply:
Unfortunately `Application.spec(:rig, :vsn)` responds `nil`… `RigApi.Mixfile.project[:version]` would work?!

Reply:
not sure we're trying the same thing here.. For me, this works:

```bash
$ iex -S mix
iex(1)> Application.spec(:rig, :vsn)
'2.0.0-beta.2'
```
```bash
$ cd apps/rig_api
$ iex -S mix
iex(1)> Application.spec(:rig, :vsn)
'2.0.0-beta.2'
```
```bash
$ mix release
$ _build/dev/rel/rig/bin/rig console
iex(rig@127.0.0.1)1> Application.spec(:rig, :vsn)
'2.0.0-beta.2'
```

The main problem with your proposed solution is that `RigApi.Mixfile` is simply not present in a release; in fact, any Mix related functionality is basically only available in dev and test.

Reply:
I did test this in the exact same way as you did. Same result, I'm getting '2.0.0-beta.2' as a response.

Still, when I use this in router.exs, like below, I'm getting a nil.
```
  def swagger_info do
    %{
      info: %{
        version: Application.spec(:rig, :vsn),
        title: "RIG Control API",
        description: "This is the description for the RIG Control API exposed on Port 4010 by default or API_PORT within the config-file.
        It manages the Proxy APIs or user connections for RIGs proxy."
      }
    }
  end
```

Reply:
Hi @kevinbader I think the reason that `Application.spec(:rig, :vsn) `returns Nil in this case is, that actually the app isn't started yet.
This is because Phoenix_swagger uses it's own mix Task `phx.swagger.generate` prior starting the actual app.

I'm not sure if this part of code is ever in use in a release setup?! Shouldn't the swagger.json only change during development & is therefore part of a commit?
Hence, Maybe it's valid to use `RigApi.Mixfile` as we only need it in dev

What's your thought's on this?

Reply:
Hi @kevinbader:

Finally got it. with `Application.ensure_all_started` the Generation of the swagger.json is forced to be executed after start. :) This works both when executing `mix compile` & `mix phx.server`

**Downside**: Application is temporary started during `mix compile`

Here's how I implemented it now & included in the latest commit :)

```
  def swagger_info do
    Application.ensure_all_started(:rig_api, :temporary) 
    %{
      info: %{
        version: Application.spec(:rig_api,:vsn) |> to_string,
        title: "RIG Control API",
        description: "This is the description for the RIG Control API exposed on Port 4010 by default or API_PORT within the config-file.
        It manages the Proxy APIs or user connections for RIGs proxy."
      }
    }
  end
```

Reply:
nice idea! I'm not really convinced that starting an application is the way to go just to obtain the release version though. Will look into this tomorrow

Reply:
@vanillebear 

seems like the compile task and the mix task do the same thing:

> PhoenixSwagger provides a mix compiler and mix task phx.swagger.generate for the swagger-ui json file generation that contains swagger specification that describes API of the phoenix application.

=> I'd remove the alias. The compile task should ensure that the file is current also when building a release.

> I'm not sure if this part of code is ever in use in a release setup?!

You're right, I got the wrong - please use what you've had initially: `RigApi.Mixfile.project[:version]` is fine since this is never executed in a release context. Sorry for the confusion.

> Shouldn't the swagger.json only change during development & is therefore part of a commit?

Isn't the JSON file a build artifact and as such out-of-scope for version control?